plugins/cassandra/cassandraembed.cpp

```diff
@@ -3718,17 +3925,41 @@ class CassMultiIterator : public CInterface, implements IRowProvider, implements
             inputs.append(*new CassSortableIterator(cass_iterator_from_result(results.item(idx)), compareColumn, descending));
         }
         merger.setown(createRowStreamMerger(inputs.length(), *this, this, false));
+        rowNum = startRowNum;
         return next();
     }
     virtual bool next()
     {
-        current.clear();
-        const CassandraIterator *nextSource = get_row();
-        if (!nextSource)
-            return false;
-        Owned<IPTree> wuXML = rowToPTree(NULL, NULL, workunitInfoMappings+1, cass_iterator_get_row(*nextSource));
-        current.setown(createConstWorkUnitInfo(*wuXML));
-        return true;
+        Owned<IConstWorkUnitInfo> last = current.getClear();
+        loop
+        {
+            const CassandraIterator *nextSource = get_row();
```

Comment:
It doesn't look like this is getting linked object.. but wondered because method name is 'get...()' which we normally associated with returning a linked object.
I think this name because associated with cassandra API method name.. but confused me a little.